packages/itwinui-react/src/core/Tooltip/Tooltip.tsx

```diff
@@ -4,10 +4,57 @@
  *--------------------------------------------------------------------------------------------*/
 import * as React from 'react';
 import cx from 'classnames';
-import { Popover, Box } from '../utils/index.js';
-import type { CommonProps, PopoverProps } from '../utils/index.js';
+import {
+  useFloating,
+  autoUpdate,
+  offset,
+  flip,
+  shift,
+  useHover,
+  useFocus,
+  useDismiss,
+  useRole,
+  useInteractions,
+  useClick,
+} from '@floating-ui/react';
+import type { Placement } from '@floating-ui/react';
+import { Box, getDocument, useGlobals } from '../utils/index.js';
+import ReactDOM from 'react-dom';
 
-export type TooltipProps = {
+type TooltipOptions = {
+  /**
+   * Placement of the Tooltip
+   * @default 'top'
+   */
+  placement?: Placement;
+  /**
+   * Property for manual visibility control
+   */
+  visible?: boolean;
+  /**
+   * Function that allows users to control Tooltip visibility
+   */
+  toggleVisible?: (open: boolean) => void;
+  /**
+   * Use this if you want Tooltip to follow moving trigger element
+   * @default false;
+   */
+  followTrigger?: boolean;
+  /**
+   *
+   */
+  middleware?: {
+    offset?: any; // eslint-disable-line @typescript-eslint/no-explicit-any
```

Comment:
Can be booleans. Or an array of string literals if you prefer.

Also should be renamed from `middleware` to `modifiers` or `options`? Because it includes more than just middleware

Reply:
also it's not used anywhere yet 😅

Reply:
Fixed